run-build

```diff
@@ -387,6 +387,27 @@ class ImageBuilder(object):
             datetime.datetime.utcnow().strftime('%y%m%d-%H%M%S')
         )
 
+    @staticmethod
+    def _get_prefixes(value):
+        """Yields all prefixes of value split by _, including the whole string.
+
+        >>> list(ImageBuilder._get_prefixes("en_GB_orkney"))
+        ["en", "en_GB", "en_GB_orkney"]
+        """
+        parts = value.split("_")
+        for i in range(1, len(parts) + 1):
+            yield "_".join(parts[:i])
+
+    def _get_attr_values(self, attr):
+        """Yields one or more values for attr. For most properties this is a single
+        value, but for "personality" this yields a series of values to implement
+        inheritance."""
+        value = getattr(self, attr)
+        if attr == "personality":
+            yield from self._get_prefixes(value)
```

Comment:
I kinda think we should do this for all fields. Certainly for product and platform it could make sense to have a specialized version that tweaks the normal version without having to abuse personality. For branch and arch it probably doesn't make sense. Just starting with personality would be fine, though.

Reply:
I originally wrote it as working for all fields. We can obviously do that but I found it easier to write the documentation & explain what the feature is for if it was just the personality field.

Reply:
Makes sense. We can always expand if it seems useful.